src/clients/pagination/iter.rs

```diff
@@ -5,6 +5,17 @@ use crate::{model::Page, ClientError, ClientResult};
 /// Alias for `Iterator<Item = T>`, since sync mode is enabled.
 pub type Paginator<'a, T> = Box<dyn Iterator<Item = T> + 'a>;
 
+pub fn paginate_with_ctx<'a, Ctx: 'a, T: 'a, Request: 'a>(
+    ctx: Ctx,
+    req: Request,
+    page_size: u32,
+) -> Paginator<'a, ClientResult<T>>
```

Comment:
I am a little confused that what's `ctx` for?

Reply:
In the sync case: not much. This is only to keep compatibility with the async version, which needs those. I'll try to explain why on the other review comments.